stan/math/rev/core/var.hpp

```diff
@@ -63,185 +67,51 @@ class var {
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
-   */
-  // NOLINTNEXTLINE
-  var(unsigned int x)
-      : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-
-  /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
-   */
-  // NOLINTNEXTLINE
-  var(unsigned long x)
-      : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-
-#ifdef _WIN64
-  // these two ctors are for Win64 to enable 64-bit signed
-  // and unsigned integers, because long and unsigned long
-  // are still 32-bit
-
-  /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
-   */
-  var(size_t x) : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-
-  /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
-   *
-   * @param x Value of the variable.
+   * Construct a variable from a pointer to a variable implementation.
+   * @param vi A vari_value pointer.
    */
-  var(ptrdiff_t x) : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-#endif
-
-#ifdef BOOST_MATH_USE_FLOAT128
-
-  // this ctor is for later GCCs that have the __float128
-  // type enabled, because it gets enabled by boost
+  var_value(vari_value<T>* vi)  // NOLINT
+      : vi_(vi) {}
 
   /**
-   * Construct a variable from the specified arithmetic argument
-   * by constructing a new <code>vari</code> with the argument
-   * cast to <code>double</code>, and a zero adjoint.
+   * Return a constant reference to the value of this variable.
    *
-   * @param x Value of the variable.
+   * @return The value of this variable.
    */
-  var(__float128 x) : vi_(new vari(static_cast<double>(x), false)) {}  // NOLINT
-
-#endif
+  inline const auto& val() const { return vi_->val_; }
 
   /**
-   * Return the value of this variable.
+   * Return a const reference of the derivative of the root expression with
+   * respect to this expression.  This method only works
+   * after one of the `grad()` methods has been
+   * called.
    *
-   * @return The value of this variable.
+   * @return Adjoint for this variable.
    */
-  inline double val() const { return vi_->val_; }
+  inline const auto& adj() const { return vi_->adj_; }
 
   /**
-   * Return the derivative of the root expression with
+   * Return a reference to the derivative of the root expression with
    * respect to this expression.  This method only works
-   * after one of the <code>grad()</code> methods has been
+   * after one of the `grad()` methods has been
```

Comment:
> This method only works...

Not totally true. The method always works, you just might not get what you expect unless you call `grad`. If you feel like rewording, do. If not, it's fine.

Reply:
If you have rewording suggestions I'm all for it!